python/cuml/common/memory_utils.py

```diff
@@ -142,6 +142,9 @@ def _order_to_strides(order, shape, dtype):
     if isinstance(shape, int):
         return (itemsize,)
 
+    elif len(shape) == 0:
```

Comment:
Why is this needed? Are you sure it's always appropriate? In numpy, it's perfectly well-defined to take the itemsize of an empty array?

Reply:
Because of this behavior in cupy and numpy: 
```python
>>> import numpy as np
>>> c = np.ones(10)
>>> d = c[0]
>>> d.__array_interface__
{'data': (93895330296096, False), 'strides': None, 'descr': [('', '<f8')], 'typestr': '<f8', 'shape': (), 'version': 3, '__ref': array(1.)}
```
that we hadn't run into before, but apparently that's how things can behave in that particular view of a single element case

Reply:
It is needed when we want to create a `CumlArray` in the case of:
```
>>> import cupy as cp
>>> x = cp.array([1,2,3])
>>> y = x[0]
>>> y
array(1)
>>> y.__cuda_array_interface__
{'shape': (), 'typestr': '<i8', 'descr': [('', '<i8')], 'version': 2, 'strides': None, 'data': (139995713110016, False)}
```
`y` has no shape here but it's a valid array still

Reply:
makes sense!